x-pack/libbeat/management/fleet/manager.go

```diff
@@ -285,3 +303,25 @@ func (cm *Manager) toConfigBlocks(cfg common.MapStr) (api.ConfigBlocks, error) {
 
 	return res, nil
 }
+
+func statusToProtoStatus(status management.Status) proto.StateObserved_Status {
```

Comment:
can we add unit test to test switching statuses back and forth with some noop client?

Reply:
At the moment I don't believe client from `elastic-agent-client` is exposed as an interface. To allow me to completely replace it with a noop client.